src/pybind/mgr/cephadm/migrations.py

```diff
@@ -0,0 +1,117 @@
+import logging
+from typing import List, TYPE_CHECKING, Iterator
+
+from ceph.deployment.service_spec import PlacementSpec, ServiceSpec, HostPlacementSpec
+from cephadm.inventory import Inventory, SpecStore, HostCache
+from cephadm.schedule import HostAssignment
+
+from orchestrator import DaemonDescription, OrchestratorError
+
+if TYPE_CHECKING:
+    from .module import CephadmOrchestrator
+
+LAST_MIGRATION = 1
```

Comment:
Feels like a poor mans way of versioning all of cephadm (e.g. v0 -> v1)?

I wonder we should add versioning to the spec instead ...

Reply:
The specs don't change in this migration. it's the scheduler here. Adding the version to the specs at this time just felt wrong to me.

Reply:
and the specs are only another part of cephadm which needs potential upgrades. We have the cache, the inventory, the config store, the upgrade (imagine changing the upgrade code, while an old upgrade is still in progress), naming of daemons, fs-layout of the daemons.

Reply:
added explanation